File: app/src/main/java/the/mic/swag/ui/fragment/posting/PostingFragment.java
package the.mic.swag.ui.fragment.posting;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.fragment.app.Fragment;

import the.mic.swag.R;

public class PostingFragment extends Fragment {

    public PostingFragment() {

    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {

        return inflater.inflate(R.layout.fragment_post, container, false);
    }
}

File: app/src/main/java/the/mic/swag/ui/fragment/posting/PostingFragment.kt
package the.mic.swag.ui.fragment.posting

import android.os.Bundle
import android.view.View
import androidx.fragment.app.FragmentManager
import androidx.lifecycle.ViewModelProviders
import com.android.dutchman.util.DataBindingFragment
import the.mic.swag.R
import the.mic.swag.databinding.FragmentPostBinding
import the.mic.swag.domain.repository.posting.PostingRepository
import the.mic.swag.presentation.viewmodel.posting.PostingViewModel
import the.mic.swag.presentation.viewmodel.posting.PostingViewModelFactory

class PostingFragment: DataBindingFragment<FragmentPostBinding>(), PostingRepository {

    override val layoutId: Int
    get() = R.layout.fragment_post

    private val factory = PostingViewModelFactory(this)

    private val fm: FragmentManager? by lazy { fragmentManager }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        val viewModel = ViewModelProviders.of(activity!!, factory).get(PostingViewModel::class.java)

        binding.vm = viewModel

    }


}